requests/models.py

```diff
@@ -363,8 +364,8 @@ def prepare_url(self, url, params):
 
         # Only want to apply IDNA to the hostname
         try:
-            host = host.encode('idna').decode('utf-8')
-        except UnicodeError:
+            host = idna.encode(host).decode('utf-8')
```

Comment:
I was testing with the sample hostname in the [idna docs](https://github.com/kjd/idna#compatibility-mapping-uts-46):

```
>>> import idna
>>> idna.encode(u'Königsgäßchen')
...
idna.core.InvalidCodepoint: Codepoint U+004B at position 1 of u'K\xf6nigsg\xe4\xdfchen' not allowed
>>> idna.encode(u'Königsgäßchen', uts46=True)
'xn--knigsgchen-b4a3dun'
```

Interestingly, it does work fine if there are not any characters that require encoding.